sklego/meta.py

```diff
@@ -117,3 +117,60 @@ def predict(self, X):
                            .loc[lambda d: d['new'] == 1]
                            .itertuples())
             raise ValueError(f"found a group(s) {culprits} in `.predict` that was not in `.fit`")
+
+
+class DecayEstimator(BaseEstimator):
+    """
+    Morphs an estimator suchs that the training weights can be
+    adapted to ensure that points that are far away have less weight.
+    Note that it is up to the user to sort the dataset appropriately.
+    This meta estimator will only work for estimators that have a
+    "sample_weights" argument in their `.fit()` method.
+
+    The DecayEstimator will use exponential decay to weight the parameters.
+
+    w_{t-1} = decay * w_{t}
+
+    """
+
+    def __init__(self, model, decay: float = 0.999):
+        self.model = model
+        self.decay = decay
+
+    def _is_classifier(self):
+        return any(['ClassifierMixin' in p.__name__ for p in type(self.model).__bases__])
+
+    def fit(self, X, y):
+        """
+        Fit the data after adapting the same weight.
+
+        :param X: array-like, shape=(n_columns, n_samples,) training data.
+        :param y: array-like, shape=(n_samples,) training data.
+        :return: Returns an instance of self.
+        """
+        X, y = check_X_y(X, y, estimator=self, dtype=FLOAT_DTYPES)
+        self.weights_ = np.cumprod(np.ones(X.shape[0]) * self.decay)[::-1]
```

Comment:
I think we don't need `weights_` as an attribute right? We don't use it in any other method

Reply:
i added it more as a convenience. say' for plotting purposes. this is especially useful when we have support for multiple decay functions.

Reply:
that's a fair point!